charts/kubernetes-agent/values.yaml

```diff
@@ -60,16 +45,40 @@ agent:
   # -- The log level of the agent. Logs are written to the pod logs as well as to file
   # @section -- Agent values
   logLevel: "Info"
-  # -- The default Kubernetes namespace for deployments
-  # @section -- Agent values
-  defaultNamespace: ""
   # -- The number of polling TCP connections to open with the target Octopus Server
   # @section -- Agent values
   pollingConnectionCount: 5
-
   # -- True if events should be scraped and added to the metrics config map
   # @section -- Agent values
   enableMetricsCapture: true
+  # -- Can be `DeploymentTarget` or `Worker`.
+  # @section -- Agent values
+  registerAs: ""
```

Comment:
I made this accept a string instead of an array of string because at the moment we would like to only allow registering as **either a worker or deployment target** but not both.

Still, I am keen on getting feedback about whether this is the right thing to do, because there is a question of whether we will end up allowing the option to register the agent as both at the same time.

If we are likely to move towards allowing both, then another option (that I prefer) is to replace this field with fields called something like `isRegistered` on child types (i.e. `deploymentTarget.isRegistered` and `worker.isRegistered`).